pages/index.js

```diff
@@ -26,9 +26,23 @@ export default function ShoppingItemsList() {
           >
             {shoppingItem.quantity} {shoppingItem.name}
             <CategoryBox>{shoppingItem.category}</CategoryBox>
+            <p>
+              <Link
+                href={{
+                  pathname: `/shoppingItems/${shoppingItem.id}`,
+                  query: {
+                    backgroundColor: shoppingItem.backgroundColor,
+                    imageURL: shoppingItem.imageUrl,
+                  },
+                }}
+              >
+                Details
+              </Link>
+            </p>
```

Comment:
- If the `shoppingItem` object already contains all the necessary data like `backgroundColor` and `imageUrl`, you don't need to pass these values through the URL query string. 

➡️ You can simply access them directly from `shoppingItem` when rendering the details page component. This would make the code cleaner and avoid potential issues related to query parameters.

- Avoid to use additional parent element solely for styling purposes. Instead, we can handle the layout and styling of the `<Link>` directly through CSS, avoiding unnecessary complexity in the HTML structure.


```suggestion
 
              <Link href={`/shoppingItems/${shoppingItem.id}`} >
                Details
              </Link>
```

Reply:
Thanks for your comments. I have changed the code accordingly, moving part of the logic into the App.js to make the props accessible in both index.js and [id].js; I have also removed the superfluous <p> tags around the links and styled the links instead.